### PR TITLE
Clarify difference between transactional() methods of Connection and EntityManager

### PR DESCRIPTION
**Pull request by sbusch**
One could interpret the old description as if `Connection#transactional()` would not rollback the transaction. Also, the fact that the `EntityManager` gets closed in case of an exception was not mentioned.

### CODE REVIEW

File: docs/en/reference/transactions-and-concurrency.rst
Comment (SenseException) [inline docs/en/reference/transactions-and-concurrency.rst:118]: Maybe this formulation is more clear. What do you think about it?
```suggestion
after the transaction rollback.
```

**Review by SenseException — CHANGES_REQUESTED** [review]

**Comment by derrabus** [comment]
@sbusch Do you want to reply to @SenseException's review?

File: docs/en/reference/transactions-and-concurrency.rst
Comment (sbusch) [reply docs/en/reference/transactions-and-concurrency.rst:118]: Sorry for late reply.

IMO there's no benefit to imply an order (EM gets closed **after** transaction rollback; is that even the correct order?) And for the user, it makes no difference for understanding the behavior. That's why I decided to write "in addition to", in contrast to the first part of the changed text where order matters.

But in the end it's a matter of taste 😃

**Comment by sbusch** [comment]
@derrabus I posted an answer. Its a matter of taste, up to you maintainers. Shouldnt block the merge

**Review by SenseException — APPROVED** [review]

**Pull request merged**
